python/dask_cudf/dask_cudf/backends.py

```diff
@@ -75,19 +75,40 @@ def _nonempty_index(idx):
     raise TypeError(f"Don't know how to handle index of type {type(idx)}")
 
 
+def _nest_list_data(data, n):
+    """
+    Helper for _get_non_empty_data which creates
+    nested list data
+    """
+    for _ in range(n):
+        data = [data]
+    return data
+
+
 @_dask_cudf_nvtx_annotate
 def _get_non_empty_data(s):
-    if isinstance(s._column, cudf.core.column.CategoricalColumn):
+    if isinstance(s, cudf.core.column.CategoricalColumn):
         categories = (
-            s._column.categories
-            if len(s._column.categories)
-            else [UNKNOWN_CATEGORIES]
+            s.categories if len(s.categories) else [UNKNOWN_CATEGORIES]
         )
         codes = cudf.core.column.full(size=2, fill_value=0, dtype="int32")
-        ordered = s._column.ordered
+        ordered = s.ordered
         data = cudf.core.column.build_categorical_column(
             categories=categories, codes=codes, ordered=ordered
         )
+    elif isinstance(s, cudf.core.column.ListColumn):
+        nesting_levels = s.dtype.nesting_levels
+        leaf_type = s.dtype.leaf_type
+        if is_string_dtype(leaf_type):
+            data = ["cat", "dog"]
+        else:
+            data = np.arange(start=0, stop=2, dtype=leaf_type).tolist()
+        data = _nest_list_data(data, nesting_levels) * 2
```

Comment:
Alternatively, you could eliminate the property `nesting_levels` and write a single function that handles this entirely in `dask_cudf`. That might avoid clutter in cudf if `nesting_levels` is only needed for this purpose. If there are other places in cudf where `nesting_levels` is needed, then the helper on the list type might be more appropriate.

```python
def _nest_list_data(data, leaf_type):
    data = [data]
    while isinstance(leaf_type, ListDtype):
        leaf_type = leaf_type.element_type
        data = [data]
    return data
```
and call
```suggestion
        data = _nest_list_data(data, leaf_type) * 2
```

Reply:
Incorporated this suggestion, I think it would be okay to make this a property if we only see a utility elsewhere too.